app/src/main/java/ar/edu/unlam/mobile/scaffolding/ui/core/component/error/ErrorHandler.kt

```diff
@@ -0,0 +1,30 @@
+package ar.edu.unlam.mobile.scaffolding.ui.core.component.error
+
+import androidx.compose.material3.SnackbarDuration
+import androidx.compose.material3.SnackbarHostState
+import androidx.compose.material3.SnackbarResult
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+
+@Composable
+fun ErrorHandler(
+    error: String?,
+    onRetry: () -> Unit,
+    snackbarHostState: SnackbarHostState,
+    onErrorShown: () -> Unit,
+) {
+    error?.let { errorMessage ->
+        LaunchedEffect(errorMessage) {
+            snackbarHostState.showSnackbar(
+                message = errorMessage,
+                actionLabel = "Reintentar",
+                duration = SnackbarDuration.Long,
+            ).let { result ->
+                if (result == SnackbarResult.ActionPerformed) {
+                    onRetry()
+                }
+                onErrorShown()
+            }
+        }
+    }
+}
```

Comment:
Podemos cambiar esta funcion por esta? Para poder determinar desde donde se llame la funcion si queremos mostrar o no mostrar el boton de retry. Por ejemplo en la pantalla de login y register veo mejor que el snackbar sea solo informativo y no tenga un reintentar, ya que el snackbar tiene una duracion corta, el usuario prefiere volver a tocar el boton de "login" o "registrar" que tocar el "reintentar" del snackbar
```suggestion
@Composable
fun ErrorHandler(
    error: String?,
    onRetry: () -> Unit,
    snackbarHostState: SnackbarHostState,
    onErrorShown: () -> Unit,
    showRetryButton: Boolean,
) {
    error?.let { errorMessage ->
        LaunchedEffect(errorMessage) {
            val snackbarResult = snackbarHostState.showSnackbar(
                message = errorMessage,
                actionLabel = if (showRetryButton) "Reintentar" else null,
                duration = SnackbarDuration.Long,
            )

            if (snackbarResult == SnackbarResult.ActionPerformed && showRetryButton) {
                onRetry()
            }

            onErrorShown()
        }
    }
}
```